### PR TITLE
Work in progress

### PR DESCRIPTION
**Pull request by dkorolev**
Hi @mzhurovich 

If you have time after https://github.com/c5t/Current/pull/238.

This PR should be safe to merge in, although it's still work in progress from my end.

* Moved to `clang-format-3.6` (to support `// clang-format off`).
* Fixed `make check`.
* Added a few static tests into `ripcurrent.h`
* On it to have typed message dispatching working.

Thanks!
Dima

### CODE REVIEW

File: RipCurrent/ripcurrent.h [hunk 1]
```diff
@@ -398,12 +404,22 @@ class NextHandlerInitializer {
 };
 
 // Base classes for user-defined code, for `is_base_of<>` `static_assert()`-s.
+template <OutputPolicy, typename EMITTABLE_TYPES_AS_TYPELIST>
+struct ConfirmRHSDoesNotEmit;
```
Comment (mzhurovich) [inline RipCurrent/ripcurrent.h]: How is it used?

**Comment by mzhurovich** [comment]
So we switch to `clang-format-3.6` as this PR is merged, right?

File: RipCurrent/ripcurrent.h (hunk 1 above)
Comment (dkorolev) [inline RipCurrent/ripcurrent.h]: It won't compile if `CURRENT_RHS` defines any type it emits.

**Comment by dkorolev** [comment]
Yes, I suggest we switch to `clang-format-3.6`. Or `alias` the 3.5 one into it, and not run `make indent` in `RipCurrent/` :)

**Pull request merged**